productivity-suite/app/legacy-app/src/components/pages/Login.tsx

```diff
@@ -16,10 +17,14 @@ export function Login() {
       return (
         getBus(ref.current).listen<{ username: string }>(
           "authentication",
-          (authDetails) => {
+          async (authDetails) => {
             if (authDetails?.username) {
-              auth.login(authDetails.username);
-              navigate(from, { replace: true });
+              await auth.login(authDetails.username);
+              const todoLists = await getTodoLists(authDetails.username);
+              getBus(ref.current!).dispatch("todo-list-selected", {
+                name: todoLists[todoLists.length - 1].name,
+              });
+              setTimeout(() => navigate(from, { replace: true }));
```

Comment:
unfortunately I needed to do this so that the navigation occurs after the `todo-list-selected`'s been properly set, otherwise the fragments would be fetched/added to the page too early